SYCL/UserDefinedReductions/user_defined_reductions_wg_size_larger_than_data_size.cpp

```diff
@@ -0,0 +1,100 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %t.out
+//
+// UNSUPPORTED: cuda || hip
+
+#include <iostream>
+#include <numeric>
+
+#include <sycl/ext/oneapi/experimental/user_defined_reductions.hpp>
+#include <sycl/sycl.hpp>
+
+template <typename T = void> struct UserDefinedMinimum {
+  T operator()(const T &lhs, const T &rhs) const {
+    return std::less<T>()(lhs, rhs) ? lhs : rhs;
+  }
+};
+
+template <> struct UserDefinedMinimum<void> {
+  struct is_transparent {};
+  template <typename T, typename U>
+  auto operator()(T &&lhs, U &&rhs) const ->
+      typename std::common_type<T &&, U &&>::type {
+    return std::less<>()(std::forward<const T>(lhs), std::forward<const U>(rhs))
+               ? std::forward<T>(lhs)
+               : std::forward<U>(rhs);
+  }
+};
+
+using namespace sycl;
+
+template <typename SegmentContainer, typename InputContainer,
+          typename OutputContainer, class BinaryOperation>
+void test(queue q, SegmentContainer segment, InputContainer input,
+          OutputContainer output, BinaryOperation binary_op,
+          const size_t num_segments, size_t workgroup_size,
+          typename OutputContainer::value_type identity,
+          typename OutputContainer::value_type init) {
+  using SegmentT = typename SegmentContainer::value_type;
+  using InputT = typename InputContainer::value_type;
+  using OutputT = typename OutputContainer::value_type;
+  constexpr size_t N = input.size();
+  {
+    buffer<InputT> seg_buf(segment.data(), segment.size());
```

Comment:
Could you please simplify the test? It seems we do not need to have a lot of segments, 1 would just enough.`
1. We could allocate an buffer of 16 elements where first 8 elements filled with `1` and the second 8 elements filled with `0`
2. submit a kernel with one wg of size 16
3. invoke `joint_reduce` with `first == ` start of the buffer and `last == ` start of the buffer + 8 elems.
4. If the patch help we should get `1` as the result, and `0` if doesn't.

Also USM allocations could be used to reduce the amount of the code.

Reply:
Done 1. - 4. Didn't change to USM because it looks like all the data initialization stuff will take more lines of code.